go.mod

```diff
@@ -14,6 +14,7 @@ require (
 	github.com/hashicorp/go-getter v1.5.5 // indirect
 	github.com/hashicorp/go-multierror v1.1.1
 	github.com/hashicorp/hcl/v2 v2.10.0 // indirect
+	github.com/hashicorp/terraform-plugin-log v0.2.0 // indirect
```

Comment:
using tflog for some diagnostic output on the new code I added. Probably should be more logging throughout?